test/disk_uploader_test.go

```diff
@@ -0,0 +1,99 @@
+package test
+
+import (
+	"context"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeclientcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	kubevirtcliv1 "kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	"kubevirt.io/kubevirt/tests/libdv"
+
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/constants"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/framework"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/runner"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/testconfigs"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/vm"
+)
+
+var _ = Describe("Run disk-uploader", func() {
+	f := framework.NewFramework()
+
+	BeforeEach(func() {
+		if f.TestOptions.SkipDiskUploaderTests {
+			Skip("skipDiskUploaderTests is set to true, skipping tests")
+		}
+	})
+
+	It("Extracts disk from VM and upload to container registry", func() {
+		vmName := constants.E2ETestsRandomName("example-vm")
+		dvName := constants.E2ETestsRandomName("example-dv")
+		secretName := constants.E2ETestsRandomName("disk-uploader-credentials")
+		imageDestination := os.Getenv("REGISTRY_IMAGE_DESTINATION")
+
+		config := &testconfigs.DiskUploaderTestConfig{
+			TaskRunTestConfig: testconfigs.TaskRunTestConfig{},
+			TaskData: testconfigs.DiskUploaderTaskData{
+				ExportSourceKind: "vm",
+				ExportSourceName: vmName,
+				VolumeName:       dvName,
+				ImageDestination: imageDestination,
+				SecretName:       secretName,
+			},
+		}
+		f.TestSetup(config)
+
+		createRegistryCredentials(f.CoreV1Client, f.DeployNamespace, secretName)
+		createAndWaitForVM(f.KubevirtClient, f.DeployNamespace, vmName, config.GetTaskRunTimeout())
+
+		runner.NewTaskRunRunner(f, config.GetTaskRun()).
+			CreateTaskRun().
+			ExpectSuccess().
+			ExpectLogs(config.GetAllExpectedLogs()...).
+			ExpectResults(nil)
+	})
+})
+
+func createRegistryCredentials(client kubeclientcorev1.CoreV1Interface, namespace, name string) {
+	registryKeyId := os.Getenv("REGISTRY_ACCESS_KEY_ID")
+	registryKey := os.Getenv("REGISTRY_SECRET_KEY")
+	v1Secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			"accessKeyId": []byte(registryKeyId),
+			"secretKey":   []byte(registryKey),
+		},
+	}
+
+	_, err := client.Secrets(namespace).Create(context.Background(), v1Secret, metav1.CreateOptions{})
+	Expect(err).ToNot(HaveOccurred())
+}
+
+func createAndWaitForVM(client kubevirtcliv1.KubevirtClient, namespace, name string, timeout time.Duration) {
+	dataVolume := libdv.NewDataVolume(
+		libdv.WithRegistryURLSource("docker://quay.io/kubevirt/cirros-container-disk-demo"),
+		libdv.WithForceBindAnnotation(),
```

Comment:
Should not be required, if the VM was started at least once.

Reply:
VM can't be exported if started and it's running. It should be stopped.

Reply:
It's created as stopped by default.